src/V3Udp.cpp

```diff
@@ -0,0 +1,235 @@
+    }
+    void visit(AstUdpTableLine* nodep) override {
+        FileLine* const fl = nodep->fileline();
+        if (!nodep->udpIsCombo() && !m_oFieldVarp->isBitLogic()) {
+            m_oFieldVarp->v3error("For sequential udp, the output var should be the reg type");
+        }
+        if (nodep->udpIsCombo() && m_oFieldVarp->isBitLogic()) {
+            m_oFieldVarp->v3error(
+                "For combinational udp, the output var shouldn't be the reg type");
+        }
+        AstNode* iNodep = nodep->iFieldp();
+        AstNode* oNodep = nodep->oFieldp();
+        uint32_t inputvars = 0;
+        AstSenTree* edgetrigp = nullptr;
+
+        AstLogAnd* logandp = new AstLogAnd(fl, new AstConst{fl, AstConst::BitTrue{}},
+                                           new AstConst{fl, AstConst::BitTrue{}});
+
+        for (AstVar* itr : m_inputVars) {
+            if (!iNodep) break;
+            inputvars++;
+            if (AstUdpTableLineVal* linevalp = VN_CAST(iNodep, UdpTableLineVal)) {
+                string valName = linevalp->name();
+                AstVarRef* const referencep = new AstVarRef{fl, itr, VAccess::READ};
+                if (isEdgeTrig(valName)) {
+                    if (nodep->udpIsCombo()) {
+                        linevalp->v3error(
+                            "There should not be a edge trigger for combinational UDP table line");
+                    }
+                    if (edgetrigp) {
+                        linevalp->v3error("There can be only one edge tigger signal");
+                    }
+                    edgetrigp = new AstSenTree{
+                        fl, new AstSenItem{fl, VEdgeType::ET_BOTHEDGE,
+                                           new AstVarRef{fl, itr, VAccess::READ}}};
+                }
+                if (valName == "0" || valName == "f")
+                    logandp = new AstLogAnd(fl, logandp, new AstLogNot(fl, referencep));
+                else if (valName == "1" || valName == "r")
+                    logandp = new AstLogAnd(fl, logandp, referencep);
+            }
+            iNodep = iNodep->nextp();
+        }
+        if (inputvars != m_inputNum) {
+            nodep->v3error("Incorrect number of input values, expected " << m_inputNum << ", got "
+                                                                         << inputvars);
+        }
+
+        string const oValName = nodep->udpIsCombo() ? oNodep->name() : oNodep->nextp()->name();
+        if (oValName == "-") {
+            if (edgetrigp) delete edgetrigp;
+            if (logandp) delete logandp;
+            return;
+        }
+
+        if (!nodep->udpIsCombo()) {
+            AstVarRef* referencep = new AstVarRef{fl, m_oFieldVarp, VAccess::READ};
+            if (oNodep->name() == "0") {
+                logandp = new AstLogAnd(fl, logandp, new AstLogNot(fl, referencep));
+            } else if (oNodep->name() == "1") {
+                logandp = new AstLogAnd(fl, logandp, referencep);
```

Comment:
Please fix other constructors also to use brace form.

```suggestion
                logandp = new AstLogAnd{fl, logandp, new AstLogNot{fl, referencep}};
            } else if (oNodep->name() == "1") {
                logandp = new AstLogAnd{fl, logandp, referencep};
```